app/frontend/src/features/auth/email/ChangeEmailPage.tsx

```diff
@@ -0,0 +1,95 @@
+import { useMediaQuery, useTheme } from "@material-ui/core";
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { Error as GrpcError } from "grpc-web";
+import { useForm } from "react-hook-form";
+import { useMutation } from "react-query";
+
+import Alert from "../../../components/Alert";
+import Button from "../../../components/Button";
+import CircularProgress from "../../../components/CircularProgress";
+import PageTitle from "../../../components/PageTitle";
+import TextField from "../../../components/TextField";
+import { service } from "../../../service";
+import useAccountInfo from "../useAccountInfo";
+import useChangeDetailsFormStyles from "../useChangeDetailsFormStyles";
+
+interface ChangeEmailFormData {
+  newEmail: string;
+  currentPassword?: string;
+}
+
+export default function ChangeEmailPage() {
+  const classes = useChangeDetailsFormStyles();
+  const theme = useTheme();
+  const isMdOrWider = useMediaQuery(theme.breakpoints.up("md"));
+
+  const {
+    data: accountInfo,
+    error: accountInfoError,
+    isLoading: isAccountInfoLoading,
+  } = useAccountInfo();
+
+  const {
+    error: changeEmailError,
+    isLoading: isChangeEmailLoading,
+    isSuccess: isChangeEmailSuccess,
+    mutate: changeEmail,
+  } = useMutation<Empty, GrpcError, ChangeEmailFormData>(
+    ({ currentPassword, newEmail }) =>
+      service.account.changeEmail(newEmail, currentPassword)
+  );
+
+  const { handleSubmit, register } = useForm<ChangeEmailFormData>();
+  const onSubmit = handleSubmit(({ currentPassword, newEmail }) => {
+    changeEmail({ currentPassword, newEmail });
+  });
+
+  return (
+    <>
+      <PageTitle>Change email</PageTitle>
+      {isAccountInfoLoading ? (
+        <CircularProgress />
+      ) : accountInfoError ? (
+        <Alert severity="error">{accountInfoError.message}</Alert>
+      ) : (
+        <>
+          {changeEmailError && (
+            <Alert severity="error">{changeEmailError.message}</Alert>
+          )}
+          {isChangeEmailSuccess && (
+            <Alert severity="success">
+              Your email change has been received. Check your new email to
+              complete the change.
+            </Alert>
+          )}
+          <form className={classes.form} onSubmit={onSubmit}>
+            {accountInfo && accountInfo.hasPassword && (
+              <TextField
+                id="currentPassword"
+                inputRef={register({ required: true })}
+                label="Current password"
+                name="currentPassword"
+                type="password"
+                fullWidth={!isMdOrWider}
+              />
+            )}
+            <TextField
+              id="newEmail"
+              inputRef={register({ required: true })}
+              label="New email"
+              name="newEmail"
+              fullWidth={!isMdOrWider}
```

Comment:
Still not sure this is ideal...

Reply:
What's your concern with it?

Reply:
Tangentially related and not sure of a good way to solve this either... but I really dislike the text field on Material UI forcing us to hardcode a width value if we don't want full width and it doesn't support letting the label taking up whatever width it needs properly. It's gonna be brittle when we eventually localise the app, when we don't know how long each translated text is gonna be...

Reply:
Just setting a visual-related field on every single field. Maybe a solution will be more obvious once the hifi is implemented. 

Re: the other comment, isn't that a problem no matter what? We could always not use the Mui field.

Reply:
Are there any text fields in the hifi?

Reply:
Had a quick look and they look more like traditional text fields - box with rounded borders, but with the label outside of the text field which will solve this problem actually.